### PR TITLE
Decouple node-problem-detector release from kubernetes

### PR DESCRIPTION
**Pull request by wangzhen127**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

This PR decouples node-problem-detector (NPD) release from kubernetes. Before, NPD version is hard coded in configure.sh and node e2e tests.This PR
- allows configuring node-problem-detector release path, version, hash and custom flags on GCI clusters (GKE and GCE)
- adds NPD cluster e2e test
- allows configuring NPD image version in node e2e test.
- bumps default NPD image version in node e2e test and fixes the test

With this PR, we can create presubmit and CI jobs for NPD PRs (https://github.com/kubernetes/node-problem-detector/issues/236). This will prevent future test breaks and make sure NPD is working correctly.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Node-Problem-Detector configuration is now decoupled from the Kubernetes release on GKE/GCE.
```

**Manual cluster test**:
Run the following command with this PR using an old version of NPD and customized release path and flags. Current NPD used is v0.6.0.
```
NODE_PROBLEM_DETECTOR_VERSION="v0.5.0" \
NODE_PROBLEM_DETECTOR_TAR_HASH="650ecfb2ae495175ee43706d0bd862a1ea7f1395" \
NODE_PROBLEM_DETECTOR_RELEASE_PATH=https://storage.googleapis.com/zhenw-gke-dev-public \
NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS="--v=2 --logtostderr --system-log-monitors=/home/kubernetes/node-problem-detector/config/kernel-monitor.json --custom-plugin-monitors=/home/kubernetes/node-problem-detector/config/kernel-monitor-counter.json --port=20256" \
kubetest --build --up \
--provider=gce \
--gcp-project=zhenw-gke-dev \
--gcp-zone=us-central1-c
```
After cluster is up, manually verify a node:
1. `kube-env` of the VM instance: environment variables are correctly set.
2. `sudo journalctl -u kube-node-installation.service`: correct NPD version is downloaded.
3. `sudo cat /etc/systemd/system/node-problem-detector.service`: NPD flags are set correctly.
4. `sudo journalctl -u node-problem-detector.service`: NPD is running fine.

**Node E2E test**:
This PR bumps the version to v0.6.2. Run the following command with this PR using a different version of NPD. 
```
make test-e2e-node FOCUS="NodeProblemDetector" SKIP="" REMOTE=true CLEANUP=true DELETE_INSTANCES=true EXTRA_ENVS="NODE_PROBLEM_DETECTOR_IMAGE=k8s.gcr.io/node-problem-detector:v0.5.0"
```
1. Verify the tests pass.
2. Verify the log that NPD v0.5.0 is used (instead of v0.6.2), which indicates that we can customize the NPD version in node e2e tests.

### CODE REVIEW

**Comment by wangzhen127** [comment]
Also added E2E tests, passing on both GKE and GCE. PTAL

File: test/e2e_node/image_list.go [hunk 1]
```diff
@@ -46,7 +46,6 @@ var NodeImageWhiteList = sets.NewString(
 	"k8s.gcr.io/stress:v1",
 	busyboxImage,
 	"k8s.gcr.io/busybox@sha256:4bdd623e848417d96127e16037743f0cd8b528c026e9175e22a84f639eca58ff",
-	"k8s.gcr.io/node-problem-detector:v0.4.1",
```
Comment (Random-Liu) [inline test/e2e_node/image_list.go]: Move `getNodeProblemDetectorImage` into a non `_test.go` file, and use it here?

**Review by Random-Liu — COMMENTED** [review]
LGTM overall.

Please split the flag change, cluster e2e change and node e2e change into 3 commits. It makes it easier to cherry-pick separately if needed in the future.

**Comment by wangzhen127** [comment]
Updated. PTAL

**Comment by Random-Liu** [comment]
@dchen1107 for top level approval.

**Comment by wangzhen127** [comment]
Just fixed the merge conflict. @Random-Liu Can you take another look?

**Comment by wangzhen127** [comment]
This change will be available in Kubernetes v1.11.10+, v1.12.8+, and v1.13.6+.

**Pull request merged**